TST xgboost tests are failing

The error occurs in our `othermlpackages` tests, and the most recent logs can be found at https://dev.azure.com/fairlearn/Fairlearn/_build?definitionId=5&_a=summary

At first glance it seems like a `FutureWarning`. This has been failing since July 5th (without any change compared to July 4th) which is roughly when pandas released a new version. Technically that was 3 days earlier, but I confirmed that the pipeline was still using the pandas version 1.2.5 until July 4th.

Since this test doesn't do anything special I'm wondering whether it's an xgboost internal issue. @riedgar-ms any thoughts?

```

        if level is not None:
>           warnings.warn(
                "Using the level keyword in DataFrame and Series aggregations is "
                "deprecated and will be removed in a future version. Use groupby "
                "instead. df.sum(level=1) should use df.groupby(level=1).sum().",
                FutureWarning,
                stacklevel=4,
            )
E           FutureWarning: Using the level keyword in DataFrame and Series aggregations is deprecated and will be removed in a future version. Use groupby instead. df.sum(level=1) should use df.groupby(level=1).sum().

/usr/share/miniconda/envs/test-conda-env/lib/python3.8/site-packages/pandas/core/generic.py:10404: FutureWarning
----------------------------- Captured stdout call -----------------------------
[08:02:14] WARNING: /tmp/build/80754af9/xgboost-split_1619724447847/work/src/learner.cc:1061: Starting in XGBoost 1.3.0, the default evaluation metric used with the objective 'binary:logistic' was changed from 'error' to 'logloss'. Explicitly set eval_metric if you'd like to restore the old behavior.
=============================== warnings summary ===============================
test_othermlpackages/test_xgboost.py::test_expgrad_classification
test_othermlpackages/test_xgboost.py::test_gridsearch_classification
test_othermlpackages/test_xgboost.py::test_thresholdoptimizer_classification
test_othermlpackages/test_xgboost.py::test_thresholdoptimizer_classification
  /usr/share/miniconda/envs/test-conda-env/lib/python3.8/site-packages/xgboost/sklearn.py:888: UserWarning: The use of label encoder in XGBClassifier is deprecated and will be removed in a future release. To remove this warning, do the following: 1) Pass option use_label_encoder=False when constructing XGBClassifier object; and 2) Encode your labels (y) as integers starting with 0, i.e. 0, 1, 2, ..., [num_class - 1].
    warnings.warn(label_encoder_deprecation_msg, UserWarning)

-- Docs: https://docs.pytest.org/en/stable/warnings.html
------------ generated xml file: /home/vsts/work/1/s/TEST-TEST.xml -------------
=========================== short test summary info ============================
FAILED test_othermlpackages/test_xgboost.py::test_expgrad_classification - Fu...
FAILED test_othermlpackages/test_xgboost.py::test_gridsearch_classification
=================== 2 failed, 1 passed, 4 warnings in 36.09s ===================
```

--- a/fairlearn/reductions/_moments/utility_parity.py
+++ b/fairlearn/reductions/_moments/utility_parity.py
@@ -248,8 +248,8 @@ class UtilityParity(ClassificationMoment):
            16-Jul-2018. [Online]. Available: https://arxiv.org/abs/1803.02453.
 
         """
-        lambda_event = (lambda_vec["+"] - self.ratio * lambda_vec["-"]).sum(level=_EVENT) / \
-            self.prob_event
+        lambda_event = (lambda_vec["+"] - self.ratio * lambda_vec["-"]) \
+            .groupby(level=_EVENT).sum() / self.prob_event
         lambda_group_event = (self.ratio * lambda_vec["+"] - lambda_vec["-"]) / \
             self.prob_group_event
         adjust = lambda_event - lambda_group_event

--- a/test_othermlpackages/test_xgboost.py
+++ b/test_othermlpackages/test_xgboost.py
@@ -10,20 +10,20 @@ xgb = pytest.importorskip("xgboost")
 
 
 def test_expgrad_classification():
-    estimator = xgb.XGBClassifier()
+    estimator = xgb.XGBClassifier(use_label_encoder=False)
     disparity_moment = DemographicParity()
 
     ptc.run_expgrad_classification(estimator, disparity_moment)
 
 
 def test_gridsearch_classification():
-    estimator = xgb.XGBClassifier()
+    estimator = xgb.XGBClassifier(use_label_encoder=False)
     disparity_moment = DemographicParity()
 
     ptc.run_gridsearch_classification(estimator, disparity_moment)
 
 
 def test_thresholdoptimizer_classification():
-    estimator = xgb.XGBClassifier()
+    estimator = xgb.XGBClassifier(use_label_encoder=False)
 
     ptc.run_thresholdoptimizer_classification(estimator)

